### PR TITLE
Allowing functions containing parameters for Contractor

### LINKED ISSUE #140: Allow parameters in Contractor
e.g. for `f(x) = r*x*(1-x)` where `r` is a parameter.

cc @yashcodes

### PR DESCRIPTION
**Pull request by yashvardhan747**
Fixes [#140](https://github.com/JuliaIntervals/IntervalConstraintProgramming.jl/issues/140)

### CODE REVIEW

**Comment by yashvardhan747** [comment]
@dpsanders as you suggested, I allowed to input parameters in function argument to avoid the problem of slowing down the process due to global variables. And also made the test cases as you needed.

File: src/contractor.jl [hunk 1]
```diff
@@ -148,7 +157,7 @@ BasicContractor(vars, f::Function) = BasicContractor([Variable(Symbol(i))() for
 
 Contractor(expr::Operation) = Contractor([], expr::Operation)
 
-Contractor(vars::Union{Vector{Operation}, Tuple{Vararg{Operation,N}}}, g::Function) where N = Contractor(vars, g(vars...)) #Contractor can be constructed by function name only
+Contractor(vars::Union{Vector{Operation}, Tuple{Vararg{Operation,N}}}, para::Union{Vector{Operation}, Tuple{Vararg{Operation,N}}}, g::Function) where N = Contractor(vars, g(vars..., para...)) #Contractor can be constructed by function name onlyContractor(vars::Union{Vector{Operation}, Tuple{Vararg{Operation,N}}}, g::Function) where N = Contractor(vars, g(vars...)) #Contractor can be constructed by function name only
```
Comment (dpsanders) [inline src/contractor.jl:160]: It should not be necessary to use so many type annotations.
Also there is some duplication at the end of the line.

Please always add a space after `#` in comments.

File: src/contractor.jl [hunk 2]
```diff
@@ -36,21 +37,25 @@ function Contractor(variables::Vector{Symbol}, top, forward, backward, expressio
...
 end
 
     (C::Contractor)(X) = C.forward(X)[1]
     (C::BasicContractor)(X) = C.forward(X)[1]
 
-function contract(C::AbstractContractor, A::IntervalBox{Nout,T}, X::IntervalBox{N,T})where {N,Nout,T}
+function contract(C::AbstractContractor, A::IntervalBox{Nout,T}, X::IntervalBox{N,T}, parameters)where {N,Nout,T}
 
+    for p in parameters
+       X = ×(X, p..p)
```
Comment (dpsanders) [inline src/contractor.jl:57]: This will be very slow. 
Also the parameters might conceivably be intervals.

File: src/contractor.jl [hunk 3]
```diff
@@ -69,25 +74,29 @@ function contract(C::AbstractContractor, A::IntervalBox{Nout,T}, X::IntervalBox{
     # @show constrained
     # @show intermediate
     # @show C.backward(X, constrained, intermediate)
-    return IntervalBox{N,T}(C.backward(X, constrained, intermediate) )
+    return IntervalBox{N,T}(C.backward(X, constrained, intermediate)[1:N])
```
Comment (dpsanders) [inline src/contractor.jl:77]: I don't understand why there's `[1:N]` here suddenly.

**Comment by yashvardhan747** [comment]
I think you can merge this PR now.

**Comment by dpsanders** [comment]
No longer relevant.

**Pull request closed without merging**